SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java

```diff
@@ -230,10 +230,30 @@ public SingularityRequest checkSingularityRequest(
       request.getId().length(),
       request.getId()
     );
-    checkBadRequest(
-      !request.getInstances().isPresent() || request.getInstances().get() > 0,
-      "Instances must be greater than 0"
-    );
+
+    if (singularityConfiguration.allowSettingRequestInstances()) {
+      checkBadRequest(
+        (
+          !existingRequest.flatMap(SingularityRequest::getInstances).isPresent() &&
+          !request.getInstances().isPresent()
+        ) ||
+        request.getInstances().get() > 0,
+        "Instances must be greater than 0"
+      );
+
+      if (
+        existingRequest.flatMap(SingularityRequest::getInstances).isPresent() &&
+        !request.getInstances().isPresent()
+      ) {
```

Comment:
quick thought, should this only apply to SERVICE/WORKER types? I think it is a valid operation to want to unset the instance count for an ON_DEMAND, effectively removing the instance count cap for that. With the validation as it stands we would no longer be able to do that